src/Dynatrace.OpenTelemetry.Exporter.Metrics.Tests/DynatraceMetricsExporterTests.cs

```diff
@@ -138,6 +139,129 @@ public async void TestExportMultipleWithPrefix()
 			Assert.Equal(expectedString, req.Content.ReadAsStringAsync().Result);
 		}
 
+		[Fact]
+		public async void TestExportMultiDataMetric()
+		{
+			var timestamp = DateTimeOffset.FromUnixTimeMilliseconds(1604660628881).UtcDateTime;
+
+			var metric = new Metric("namespace1", "metric1", "Description", AggregationType.LongSum);
+			metric.Data.Add(new Int64SumData
+			{
+				Sum = 100,
+				Timestamp = timestamp
+			});
+			metric.Data.Add(new Int64SumData
+			{
+				Sum = 101,
+				Timestamp = timestamp
+			});
+
+			var mockMessageHandler = new Mock<HttpMessageHandler>();
+			HttpRequestMessage req = null;
+
+			mockMessageHandler.Protected()
+			.Setup<Task<HttpResponseMessage>>("SendAsync", ItExpr.IsAny<HttpRequestMessage>(), ItExpr.IsAny<CancellationToken>())
+			.ReturnsAsync(new HttpResponseMessage
+			{
+				StatusCode = HttpStatusCode.OK,
+				Content = new StringContent("test")
+			})
+			.Callback((HttpRequestMessage r, CancellationToken _) => req = r);
+
+			var exporter = new DynatraceMetricsExporter(null, null, new HttpClient(mockMessageHandler.Object));
+
+			await exporter.ExportAsync(new List<Metric> { metric }, CancellationToken.None);
+			var expectedString = "namespace1.metric1,dt.metrics.source=opentelemetry count,delta=100 1604660628881" + Environment.NewLine +
+								 "namespace1.metric1,dt.metrics.source=opentelemetry count,delta=101 1604660628881" + Environment.NewLine;
+
+			mockMessageHandler.Protected().Verify("SendAsync", Times.Exactly(1), ItExpr.Is<HttpRequestMessage>(req => req.Method == HttpMethod.Post), ItExpr.IsAny<CancellationToken>());
+			Assert.Equal(expectedString, req.Content.ReadAsStringAsync().Result);
+		}
+
+		[Fact]
+		public async void TestExportNullNameMetric()
+		{
+			var timestamp = DateTimeOffset.FromUnixTimeMilliseconds(1604660628881).UtcDateTime;
+
+			var metric = new Metric(null, null, null, AggregationType.LongSum);
+			metric.Data.Add(new Int64SumData
+			{
+				Sum = 100,
+				Timestamp = timestamp
+			});
+
+			var mockMessageHandler = new Mock<HttpMessageHandler>();
+			HttpRequestMessage req = null;
+
+			mockMessageHandler.Protected()
+			.Setup<Task<HttpResponseMessage>>("SendAsync", ItExpr.IsAny<HttpRequestMessage>(), ItExpr.IsAny<CancellationToken>())
+			.ReturnsAsync(new HttpResponseMessage
+			{
+				StatusCode = HttpStatusCode.OK,
+				Content = new StringContent("test")
+			})
+			.Callback((HttpRequestMessage r, CancellationToken _) => req = r);
+
+			var mockLogger = new Mock<ILogger<DynatraceMetricsExporter>>();
+
+			var exporter = new DynatraceMetricsExporter(null, mockLogger.Object, new HttpClient(mockMessageHandler.Object));
+
+			await exporter.ExportAsync(new List<Metric> { metric }, CancellationToken.None);
+			var expectedString = String.Empty;
+
+			mockMessageHandler.Protected().Verify("SendAsync", Times.Exactly(1), ItExpr.Is<HttpRequestMessage>(req => req.Method == HttpMethod.Post), ItExpr.IsAny<CancellationToken>());
```

Comment:
should it really send a request even if there are no valid metrics?

Reply:
You're right, thought the same thing but wanted to keep the original behavior.
I added a check in `DynatraceMetricsExporter` in [844fd4e](https://github.com/dynatrace-oss/opentelemetry-metric-dotnet/pull/24/commits/844fd4e7b3d76bd088e669481ab057509e5b4ad0) and adapted the tests.